wip-mariocao-median.md

```diff
@@ -0,0 +1,98 @@
+<pre>
+  WIP: WIP-mariocao-median
+  Layer: Consensus (hard fork)
+  Title: Add average median to RADON reducers
+  Authors: Mario Cao <mario@witnet.foundation>
+  Discussions-To: `#dev-general` channel on Witnet Community's Discord server
+  Status: Draft
+  Type: Standards Track
+  Created: 2020-09-01
+  License: BSD-2-Clause
+</pre>
+
+
+## Abstract
+
+The **average median** is proposed to be included in the reducers catalog available in the RAD engine. Data requests will then support median function as part of the **retrieval**, **aggregation** and **tally** scripts.
+
+
+## Motivation and rationale
+
+It is paramount to continue increasing the catalog of available reducer functions for data requests within the RAD engine to increase the flexibility and potential of supporting additional use cases by the Witnet protocol.
+
+This document proposes to add the support of the **Median** as part of the `reducer` functions because, depending on the use case, it may be more descriptive than the average mean.
+
+Specifically for the Witnet protocol, it is interesting because of the following reasons:
+ 1. It mitigates the *negative* effect that outliers may have in a data request result at any stage where multiple data points are aggregated. For example, a data request using the **mean** function is sensitive to tampered with if malicious node reports are not filtered out previously.
+ 2. It can is advantageous with the commit/reveal scheme of the data request lifecycle. Malicious nodes cannot affect a result unless they take over most of the data request committee.
+
+
+## Specification
+
+This document proposes to add the support of the **Median** to the available RADON reducer functions.
+
+The *Median* is defined as the "middle number" in a sorted, ascending or descending, list of numbers. If the data set has an odd number of observations, the number in the middle of the list is the median. Else, if data set has an even number of observations, then the median is defined as the arithmetic mean of the two middle values.
```

Comment:
```suggestion
The *Median* is defined as the "middle number" in a sorted, ascending or descending, list of numbers. If the data set has an odd number of observations, the number in the middle of the list is the median. Otherwise, if the data set has an even number of observations, then the median is defined as the arithmetic mean of the two middle values.
```